### ISSUE TITLE
MethodOverloading: False positive when overriding from external parent class/interface

### ISSUE THREAD

**Issue by bsara** [detekt/detekt#3408]
## Expected Behavior
If the threshold for MethodOverloading is `3`, the following class should be seen as valid because the methods are overridden from the parent class:

```kotlin
class MyClass : ParentClass {
  override fun myFunc(): String = //...
  override fun myFunc(param0: String, param1: String): String = //...
  override fun myFunc(param0: Int, param1: String): String = //...
}
```

## Observed Behavior
Class in code above is seen as not valid regardless of the fact that the functions are overrides.

## Steps to Reproduce
1. Create a Java class or interface with some overloaded methods that can be overridden.
2. Create a Kotlin class that inherits the Java class/interface from step 1 and override the overloaded methods created.
3. Set the threshold of `MethodOverloading` rule to equal the number of overloaded methods that are being overridden in the Kotlin class.

## Context
I am implementing a custom builder from a base class provided by an external library that is written in Java.

## Your Environment

* Version of detekt used: 1.15.0
* Version of Gradle used (if applicable): 6.6.1
* Operating System and version: macOS Big Sur (version 11.1)
* Type Referencing: Enabled

**Comment by schalkms** [comment]
This is expected behavior. Even though the class is implementing functions from an external class, the excessive use of method overloading pertains. It's out of scope of detekt to reliable detect the usage of an external class library.

**Comment by schalkms** [comment]
The usage of an Suppress annotation is also fine here.

**Comment by bsara** [comment]
doesn't it seem strange that something that is out of my control is being flagged as invalid though?

**Comment by schalkms** [comment]
If you use 3rd party code, that's entirely possible for static analyzers. The static analyzer doesn't make a difference between own and 3rd party code.
